client/src/ui/NavBar.tsx

```diff
@@ -1,44 +1,111 @@
-import { Typography } from "@mui/material";
+"use client";
 import Link from "next/link";
+import { Typography } from "@mui/material";
+import { usePathname } from "next/navigation";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faMessage,
+  faChartLine,
+  faMagnifyingGlass,
+} from "@fortawesome/free-solid-svg-icons";
+import { NavbarMenuDropdown } from "./NavbarMenuDropdown";
+import { LogoNameLink } from "./LogoNameLink";
 
 interface ILinks {
   name: string;
   href: string;
+  icon: JSX.Element;
 }
 export function NavBar() {
   const links: ILinks[] = [
     {
       name: "Explore",
       href: "/explore",
+      icon: (
+        <FontAwesomeIcon
```

Comment:
I wouldn't store the actual component. What you can do instead is add the Icon (ex: icon: faMagnifyingGlass). Then you map it over the actual FontAwesomeIcon component. This reduces redundancy of creating the each component in the array